Filtered_kernel/include/CGAL/Filtered_predicate.h

```diff
@@ -111,6 +113,77 @@ Filtered_predicate<EP,AP,C2E,C2A,Protection>::
     return ep(c2e(args)...);
 }
 
+template <class EP_RT, class EP_FT, class AP, class C2E_RT, class C2E_FT, class C2A, bool Protection = true>
+class Filtered_predicate_RT_FT
+{
+  C2E_RT c2e_rt;
+  C2E_FT c2e_ft;
+  C2A c2a;
+  EP_RT ep_rt;
+  EP_FT ep_ft;
+  AP ap;
+
+  using Ares = typename AP::result_type;
+
+public:
+  using result_type =  typename EP_FT::result_type;
+
+  template <typename... Args>
+  struct Call_operator_needs_FT {
+    // This type traits class checks if the call operator can be called with
+    // `(const Args&..., RT_sufficient())`.
+    using ArrayOfOne = char[1];
+    using ArrayOfTwo = char[2];
+
+    static ArrayOfOne& test(...);
+
+    template <typename... Args2>
+    static auto test(const Args2 &...args)
+        -> decltype(ap(c2a(args)..., RT_sufficient()),
+                    std::declval<ArrayOfTwo &>());
+
+    enum { value = sizeof(test(std::declval<const Args&>()...)) == sizeof(ArrayOfOne) };
+  };
+
+  // ## Important note
+  //
+  // If you want to remove of rename that member function template `needs_ft`,
+  // please also change the lines with
+  // `CGAL_GENERATE_MEMBER_DETECTOR(needs_ft);`
+  // or `has_needs_ft<typename R::Compare_distance_3>` in
+  // the file `Kernel_23/test/Kernel_23/include/CGAL/_test_new_3.h`.
+  template <typename... Args>
+  bool needs_ft(const Args&...) const {
```

Comment:
constexpr?

Reply:
We can add `constexpr`, and I will... but I do not think that will be useful, or used.

That `constexpr` qualifier would be useful for something like:
```c++
  constexpr b = compare_distance_functor.needs_ft(arg1, arg2, arg3);
```
**if** both `compare_distance_functor` and all three arguments are constant expressions.

I could also make that function `static`, to remove the implicit `this` argument, but I would need another version of `has_needs_ft` for the test in `Kernel_23/test/Kernel_23/include/CGAL/_test_new_3.h`. Given that this function template `needs_ft` is only used for a few statements in a test, I do not want to work more on it.

Reply:
Done in https://github.com/CGAL/cgal/pull/6836/commits/8521c44c6f7ec9bb0096b3b74af4547ef41bf1c4.